packages/runtime/src/runtimes/react/controls/style-v2.ts

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useId } from 'react'
+import {
+  ControlDefinition,
+  StyleV2Control,
+  StyleV2ControlData,
+  StyleV2ControlDefinition,
+  StyleV2Format,
+} from '../../../controls'
+import { useStyle } from '../use-style'
+import { BoxModel, getBox } from 'css-box-model'
+import deepEqual from '../../../utils/deepEqual'
+import {
+  findBreakpointOverride,
+  getBaseBreakpoint,
+  getBreakpointMediaQuery,
+} from '../../../state/modules/breakpoints'
+import { useBreakpoints } from '..'
+import { CSSObject } from '@emotion/css'
+
+function useStyleControlCssObject(
+  styleControlData: StyleV2ControlData,
+  controlDefinition: StyleV2ControlDefinition,
+): CSSObject {
+  const breakpoints = useBreakpoints()
+
+  return {
+    ...controlDefinition.config.getStyle(
+      findBreakpointOverride(breakpoints, styleControlData, getBaseBreakpoint(breakpoints).id)
+        ?.value,
+    ),
+    ...breakpoints.reduce((styles, breakpoint) => {
+      return {
+        ...styles,
+        [getBreakpointMediaQuery(breakpoint)]: controlDefinition.config.getStyle(
+          findBreakpointOverride(breakpoints, styleControlData, breakpoint.id)?.value,
+        ),
+      }
+    }, {}),
+  }
+}
+
+export type StyleV2ControlFormattedValue = CSSObject | string
+
+export type ResolveStyleV2ControlFormattedValue<T extends ControlDefinition> =
```

Comment:
This `ResolveXXXXXXXX` is something I saw in the Image control for Image format, but the `ControlValue` of the image control doesn't seem to be typed based on which format you use in `register-components.ts`, I remember bringing this up when I was working on www.makeswift.com. I might have enough context to understand why.

Why does `useImageControlValue` return `ImageControlValue` instead of `ResolveImageControlValue`
![Uploading CleanShot 2023-05-23 at 15.17.11@2x.png…]()

Reply:
> Why does useImageControlValue return ImageControlValue instead of ResolveImageControlValue

Seems to be a mistake